src/pages/visualization-view/visualization-view.js

```diff
@@ -21,26 +21,46 @@ const VisualizationView = (props) => {
     reset,
     load,
     save,
+    update, 
     id: idFromState,
     query,
+    title,
     isLoadFailure,
     isSaving,
     isSaveSuccess,
     isSaveFailure,
     isSaveable,
+    updateQuery,
     match: {params: {id: idFromUrl}},
     history
   } = props
 
   const linkUrl = idFromState && generatePath(routes.visualizationView, {id: idFromState})
   const [isDialogOpen, setDialogOpen] = useState(false)
+  const [queryTitle, setQueryTitle] = useState(title || '')
+  const shouldUpdateQuery = updateQuery || props.isSaving
 
   React.useEffect(() => { reset() }, [])
   React.useEffect(() => { if (idFromUrl) load(idFromUrl) }, [idFromUrl])
   React.useEffect(() => { if (idFromState) history.push(linkUrl) }, [idFromState])
 
-  const handleSave = () => { setDialogOpen(true); save('Placeholder Title', query) }
-  const closeDialog = () => { setDialogOpen(false) }
+
+  const handleTitleChange = (event) => {setQueryTitle(event.target.value)}
+  const handleUpdate = () =>  {
+    setDialogOpen(true)
+    if(shouldUpdateQuery) {
+      update(idFromState, title, query)
```

Comment:
So, wait, we do the update here and if they click save? - Ben and Tim

Reply:
we renamed it to make it clear and removed the update